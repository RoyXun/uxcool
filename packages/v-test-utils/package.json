{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
