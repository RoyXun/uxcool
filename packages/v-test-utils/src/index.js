import $ from 'jquery';

export * from '@vue/test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}
