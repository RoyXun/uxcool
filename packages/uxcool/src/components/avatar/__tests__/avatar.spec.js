import { mountPickerFactory, waitTime, triggerEvent } from '@suning/v-test-utils';
import Avatar from '../index';

describe('avatar', () => {
  const mountAvatar = mountPickerFactory(Avatar);
  it('render string correctly', async () => {
    const TEST_STRING = 'test string';
    const wrapper = await mountAvatar({
      slots: {
        default: TEST_STRING,
      },
    });
    expect(wrapper.element.children.length).toBe(1);
    expect(wrapper.element.children[0].innerHTML).toBe(TEST_STRING);
  });

  it('use fallback string corectly on get img fail', async () => {
    const TEST_STRING = 'IMG_FAIL';
    const CORRECT_URL = 'https://image4.suning.cn/uimg/cms/img/155540698836163295.png';
    const ERROR_URL = 'http://error.url';
    const wrapper = await mountAvatar({
      propsData: {
        src: CORRECT_URL,
      },
      slots: {
        default: TEST_STRING,
      },
    });
    expect(wrapper.element.children.length).toBe(1);
    expect(wrapper.element.children[0].tagName).toBe('IMG');
    expect(wrapper.element.children[0].getAttribute('src')).toBe(CORRECT_URL);
    wrapper.setProps({ src: ERROR_URL });
    await waitTime();
    await triggerEvent(wrapper.find('img'), 'error');
    expect(wrapper.element.children.length).toBe(1);
    expect(wrapper.element.children[0].tagName).toBe('SPAN');
    expect(wrapper.element.children[0].innerHTML).toBe(TEST_STRING);
  });
});
