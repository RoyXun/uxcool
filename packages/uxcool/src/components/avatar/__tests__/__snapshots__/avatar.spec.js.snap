// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar render different sizes correctly 1`] = `
<span
  class="ux-avatar ux-avatar-sm ux-avatar-circle ux-avatar-icon"
>
  <i
    class="fu fu-account"
  />
</span>
`;

exports[`avatar render different sizes correctly 2`] = `
<span
  class="ux-avatar ux-avatar-circle ux-avatar-icon"
>
  <i
    class="fu fu-account"
  />
</span>
`;

exports[`avatar render different sizes correctly 3`] = `
<span
  class="ux-avatar ux-avatar-lg ux-avatar-circle ux-avatar-icon"
>
  <i
    class="fu fu-account"
  />
</span>
`;

exports[`avatar render scale correctly 1`] = `
<span
  class="ux-avatar ux-avatar-circle"
>
  <span
    style="position: absolute; display: inline-block; transform: scale(0.9411764705882353);"
  >
    Abcde
  </span>
</span>
`;
